packages/lib/prometheus.js

```diff
@@ -93,29 +93,48 @@ function labelsToKey(labels, metricLabels) {
 function keyToLabels(key) {
 	let labels = new Map();
 
-	if (key !== "") for (let pair of key.split(",")) {
-		let [name, value] = pair.split("=", 2);
-		labels.set(name, value
-			.slice(1, -1)
-			.replace(/\\"/g, "\"")
-			.replace(/\\n/g, "\n")
-			.replace(/\\\\/g, "\\")
-		);
+	if (key !== "") {
+		for (let pair of key.split(",")) {
+			let [name, value] = pair.split("=", 2);
+			labels.set(name, value
```

Comment:
I wonder if this can be done as a single RegExp or just run multiple replace commands without the regexp engine?

Reply:
If you use a string instead of a RegExp with the global flag only the first occurrence is replaced, so that's not an option.  It's possible to use a single RegExp with a replacement function to select the appropriate replacement instead of a string though.

Reply:
Well, it's up to you if you wanna replace it with single RegExp. It's just initialising and running even a single RegExp is quite a heavy operation and we have 3 here.